newlib/libc/stdio/nano-vfprintf.c

```diff
@@ -167,6 +167,7 @@ static char *rcsid = "$Id$";
 #include "fvwrite.h"
 #include "vfieeefp.h"
 #include "nano-vfprintf_local.h"
+#include "../machine/xtensa/pgmspace.h"
```

Comment:
General note, for all 4 MRs: common (non-machine-specific) code in newlib typically does not include machine-specific files directly.
Instead, what you can do is provide dummy definitions of pgm_read_byte in stdio/string.h, which will work for all platforms, and then add machine-specific definition of pgm_read_byte to sys/string.h under xtensa machine (path in this repository would be newlib/libc/sys/xtensa/sys/string.h).

Reply:
I was trying to avoid any chance of namespace collisions by putting it into a directory that is not installed to either the SDK or Arduino SDK, and to minimize duplication.  Wouldn't placing it there end up causing there to be two definitions in the tree (one in the SDK, one in the libc)?  It's not a big deal to rename, but then you've got two identical functions with the same name in a real build tree, no?

Reply:
Since we are adding these functions into libc, we can remove them from Arduino.